### ISSUE TITLE
Inverted x/y for transformation from WSG84 to LV95 (EPSG:2056)

### ISSUE THREAD

**Issue by turbotimon** [pyproj4/pyproj#1377]
#### Code Sample, a copy-pastable example if possible

When transforming from WSG84 to LV95, the coordinates get inverted

```python
from pyproj import Transformer

WGS84_CRS = 'EPSG:4326'  # WGS84 coordinate reference system
LV95_CRS = 'EPSG:2056'    # LV95 (Swiss Grid) coordinate reference system

lon_x, lat_y = 8.5417, 47.3769 # Bern, Switzerland

transformer = Transformer.from_crs(WGS84_CRS, LV95_CRS)
lv95_x, lv95_y = transformer.transform(lon_x, lat_y)
print(lv95_x, lv95_y) # 7647764, -1894770 wrong

lv95_x, lv95_y = transformer.transform(lat_y, lon_x) # inverting lon/lat
print(lv95_x, lv95_y) # 2683303, 1247925 correct
```

#### Problem description

All inputs and output should be X/Y or LON/LAT order since pyrpoj >2.6

#### Expected Output

- 2683303, 1247925
- see https://epsg.io/transform#s_srs=4326&t_srs=2056&x=8.5417000&y=47.3769000

#### Environment Information

 - Output from: `pyproj -v`
```
pyproj info:
    pyproj: 3.6.1
      PROJ: 9.3.0
  data dir: /home/erti/PROJECT_repos/OsmCompleteness/osm-completeness/venv/lib/python3.9/site-packages/pyproj/proj_dir/share/proj
user_data_dir: /home/erti/.local/share/proj
PROJ DATA (recommended version): 1.15
PROJ Database: 1.2
EPSG Database: v10.094 [2023-08-08]
ESRI Database: ArcGIS Pro 3.1 [2023-19-01]
IGNF Database: 3.1.0 [2019-05-24]

System:
    python: 3.9.18 (main, Aug 24 2023, 18:16:58)  [GCC 12.3.0]
executable: /home/erti/PROJECT_repos/OsmCompleteness/osm-completeness/venv/bin/python
   machine: Linux-6.1.77-x86_64-with-glibc2.38

Python deps:
   certifi: 2023.11.17
    Cython: None
setuptools: 69.0.2
       pip: 23.3.1
```
#### Installation method
 - pip

**Comment by djhoese** [comment]
https://pyproj4.github.io/pyproj/stable/gotchas.html#axis-order-changes-in-proj-6

For your specific case, EPSG 4326 has an axis order of (y, x) so you need to pass the inputs as lat first, lon second. You can pass `always_xy=True` to your Transformer creation which will always expect x (lon) as the first argument and y (lat) as the second argument to the transformation methods.

**Comment by turbotimon** [comment]
Ok sorry then. I supposed that coordinates are always the same order since is `transform(xx, yy, ...)`. That seems really confusing to pass lat as x coordinate..

Also, because https://pyproj4.github.io/pyproj/stable/api/proj.html#pyproj-proj says
> Converts from **longitude, latitude** to native map projection **x,y** coordinates and vice versa

**Comment by djhoese** [comment]
Right. Overall `Proj` is deprecated in favor of `Transformer`. Any function signature that uses `x` and `y` is just unfortunate naming. The variables have to be named something. Besides the `Proj` class, if there is something else in the docs that would have revealed this sooner feel free to create a pull request with the updates.
